CMakeLists.txt

```diff
@@ -164,6 +164,20 @@ option(FINAL_BUILD "Build as single source file." OFF)
 
 option(FFT_GREEN "Use internal 'Green' FFT lib rather than FFTW. (Not recommended.)" OFF)
 
+# some compilers (e.g. MSVC) don't set the usual SSE related definitions,
+# so we have to set our own
+
+cmake_host_system_information(RESULT SC_SYSTEM_HAS_SSE QUERY HAS_SSE)
+cmake_host_system_information(RESULT SC_SYSTEM_HAS_SSE2 QUERY HAS_SSE2)
```

Comment:
every x86 that doesn't have sse1/sse2 should probably go to a museum (especially they are available on all 64bit intel CPUs)

it might be better to port the feature detection macros to `boost.predef` to ensure portability:
https://www.boost.org/doc/libs/1_66_0/doc/html/predef/reference.html#predef.reference.boost_hw_hardware_macros